lix/source-code/client/src/isomorphic-git-forks/git-fetch-request-helpers.ts

```diff
@@ -0,0 +1,84 @@
+function padHex(pad: number, n: number) {
+    const s = n.toString(16);
+    return '0'.repeat(pad - s.length) + s;
+}
+
+/**
+ * Takes a line and addes the line lenght as 4 digit fixed hex value at the beginning…
+ *
+ * @param line a line raw used in a git-upload-pack request like: "want d7e62aef79d771d1771cb44c9e01faa4b7a607fe multi_ack_detailed no-done side-band-64k ofs-delta agent=git/isomorphic-git@1.24.5 filter"
+ * @returns the line enriched with the amount of characters hex encoded in the first 4 characters:
+ * "want d7e62aef79d771d1771cb44c9e01faa4b7a607fe multi_ack_detailed no-done side-band-64k ofs-delta agent=git/isomorphic-git@1.24.5 filter\n"
+ * becomes
+ * "008cwant d7e62aef79d771d1771cb44c9e01faa4b7a607fe multi_ack_detailed no-done side-band-64k ofs-delta agent=git/isomorphic-git@1.24.5 filter\n"
+ *
+ * this is because the prefixed hex 008c -> dec 140 and the leght of the line is 136 + 4 length characters. The only line that doesn't take the
+ * length characters into account is the flush line. this is signalled by 0000
+ */
+export function encodePkLine(line: string) {
+
+    const flushLine = ""
+
+    if (line === flushLine) {
+        // flush lines don't take the padded hex into account length = 0 (ignoring the leading 0000)
+        return Buffer.from(padHex(4, 0), 'utf8');
```

Comment:
btw. not sure why you use Buffer here, iso git removed using buffers in its api, could you explain? the iso git http client also does not use the old node buffer plyfill.